### DISCUSSION TITLE
Isn't the presence of settings.xml in the project root confusing?

### DISCUSSION THREAD

**Discussion by PavelTurk** [apache/netbeans#7917] [General]
### Apache NetBeans version

Apache NetBeans 23

### What happened

This is my screenshot
![Screenshot from 2024-10-28 10-26-59](https://github.com/user-attachments/assets/0fee92a4-43d8-463e-b190-5705b22d1263)

`Settings.xml` is maven `.m2/settings.xml`. I think that showing this file in the project was done intentionally. But isn't it confusing because project does NOT contain `settings.xml` file, so why is it shown in project tree? Besides the same file is shown in all projects.

### How to reproduce

Create `.m2/settings.xml` file and use it in maven plugins.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Ubuntu 20.04.6 LTS

### JDK

Java: 22.0.1; OpenJDK 64-Bit Server VM 22.0.1+8-16

### Apache NetBeans packaging

Apache NetBeans binary zip

### Are you willing to submit a pull request?

No

**Comment by matthiasblaesing** [comment]
From my POV this is not a bug, but a feature. The global `settings.xml` can have a direct impact on the users build of a project. For example I have to adjust/override the lombok version, as it typically lacks behind in projects.

What is the problem you are seeing?

**Comment by PavelTurk** [comment]
@matthiasblaesing I've explained it in my issue. Shortly two problems - 1) in `Project files` I expect to see **project** files, but not files from somewhere else. 2) all all projects and their child projects, and the child projects of the child projects :) show me the same file.

For example, my project uses git, but gitconfig is NOT shown in every project.

**Reply by matthiasblaesing** [reply]
As I said, for me its a feature and thus is a discussion, not a bug. You are looking that the _project_ view of a _maven_ project. And in this context the `settings.xml` is relevant. What you are expecting you can find in the `Files` view. That is the unprojected FS view.

**Comment by mbien** [comment]
The `Projects` tab does not necessarily show the physical layout of a project. The `Files` tab is for browsing the project folder.

It makes sense to me to have settings.xml close to pom.xml, since both are closely related and your project is affected by settings.xml.

"Dependencies" are also not part of your project folder. It is a computed list of project dependencies, the physical location is somewhere else.

**Comment by PavelTurk** [comment]
@mbien Please, note, that I was talking about `Project Files` node, but not about `Projects` tab.

**Reply by Chris2011** [reply]
I see your point and sometimes I came also over the wording. In ant and HTML projects, we call it "important files". Maybe a little rewording would be better? Also be consistent with each other? Does this make sense?

**Comment by PavelTurk** [comment]


**Reply by PavelTurk** [reply]
@neilcsmith-net I see that you don't agree with my comment. Could you then answer a simple question - What files are shown in `Project Files` node of maven project tree? I mean in the node that the red arrow points to:

![Screenshot from 2024-10-28 10-26-59](https://github.com/user-attachments/assets/6d84201e-64c5-419f-baf4-259d8df27ecc)

**Reply by neilcsmith-net** [reply]
@PavelTurk I don't agree with your comment but didn't want to get involved in the discussion given others have already pretty much covered my viewpoint.  Think of `Project Files` as `Project Configuration` and maybe it makes more sense?  I would personally -1 removing `settings.xml` from the project tree, and my :-1: reflected that.  But perhaps some clarification of what is here and where they're located (if external) would be good.

The parent POM is already reflected if you use the Effective POM view on the POM file, which is rather more complex than you describe and more difficult to convey in the project tree.

**Reply by neilcsmith-net** [reply]
Just open the POM and look at the Effective tab in a typical project.  Do you include parents up to inherited values from Maven or imported BOMs, etc.?  Unlike direct access to `settings.xml` I'm not convinced trying to expose the hierarchy in the tree is useful.  It's not like we start exposing direct access to superclasses and interfaces on Java files!

**Reply by PavelTurk** [reply]
@neilcsmith-net 
Thank you for clarifying. I know about the Effective tab, which is why I asked for an explanation. On the Effective tab, as I understand it, we see the final pom, including all parents and even Maven’s default settings. 

But we're talking about informing the user about the list of files that affect these final settings and giving them the ability to work with these files. The idea is for the person to look in `Project Files` or `Project Configurations` and immediately understand, for instance, "okay, I have totally 3 pom files." This was exactly the intention of the developers when they added `settings.xml` there—to make sure the user is aware of this file and can quickly open it.

That's why it seems strange to me – some files are shown, and some are not. Or maybe I’m the only one who finds this strange

**Comment by wumpz** [comment]
IMHO, same applies to **nbactions.xml**, **nb-configuration.xml**. There was another discussion about redesigning the project tab, which went too far.
However, I would like to have this project view more configurable. For instance, to move Dependencies and Test Dependencies into "Project Files". A standard maven project has one folder (src) (I know, files Tab). I do not count the target dir. And Netbeans extracts from this a view of 8 folders.  Another open request, is to put some special files (Jenkinsfile) into Project Files as well.

**Reply by neilcsmith-net** [reply]
@wumpz hiding direct access to `nbactions.xml` and `nb-configuration.xml` sounds even worse!

**Reply by wumpz** [reply]
Not hiding: restructuring since pom.xml is base configuration for maven project it should be prominent. But those nb -files are only netbeans specific additions to those projects. However, thinking about this, I have access to those files using the files tab. Why do I need it in the project tab? I can use the project option dialog to modify those file. 

So why is it worse? IMHO its better to show less in project view.

What does a developer need to develop / work on a module? And to what extend? Most of the time I need

pom.xml
source packages
other sources
test packages
test sources
